lib/application.js

```diff
@@ -369,7 +371,7 @@ function camelize(str) {
   return stringUtils.camelize(str);
 }
 
-function dataSourcesFromConfig(config, connectorRegistry) {
+function dataSourcesFromConfig(name, config, connectorRegistry, registry) {
```

Comment:
What is the purpose of the new `name` parameter? AFAICT, it's not used anywhere.